python/lsst/daf/butler/remote_butler/_http_connection.py

```diff
@@ -0,0 +1,226 @@
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This software is dual licensed under the GNU General Public License and also
+# under a 3-clause BSD license. Recipients may choose which of these licenses
+# to use; please see the files gpl-3.0.txt and/or bsd_license.txt,
+# respectively.  If you choose the GPL option then the following text applies
+# (but note that there is still no warranty even if you opt for BSD instead):
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("RemoteButlerHttpConnection", "parse_model")
+
+from collections.abc import Mapping
+from typing import TypeVar
+from uuid import uuid4
+
+import httpx
+from lsst.daf.butler import __version__
+from pydantic import BaseModel, ValidationError
+
+from .._exceptions import create_butler_user_error
+from ._authentication import get_authentication_headers
+from .server_models import CLIENT_REQUEST_ID_HEADER_NAME, ERROR_STATUS_CODE, ErrorResponseModel
+
+_AnyPydanticModel = TypeVar("_AnyPydanticModel", bound=BaseModel)
+"""Generic type variable that accepts any Pydantic model class."""
+
+
+class RemoteButlerHttpConnection:
+    """HTTP connection to a Butler server.
+
+    Parameters
+    ----------
+    http_client : `httpx.Client`
+        HTTP connection pool we will use to connect to the server.
+    server_url : `str`
+        URL of the Butler server we will connect to.
+    access_token : `str`
+        Rubin Science Platform Gafaelfawr access token that will be used to
+        authenticate with the server.
+    """
+
+    def __init__(self, http_client: httpx.Client, server_url: str, access_token: str) -> None:
+        self._client = http_client
+        self.server_url = server_url
+        self._access_token = access_token
+
+        auth_headers = get_authentication_headers(access_token)
+        headers = {"user-agent": f"RemoteButler/{__version__}"}
+
+        self._headers = auth_headers | headers
+
+    def post(self, path: str, model: BaseModel) -> httpx.Response:
+        """Send a POST request to the Butler server.
+
+        Parameters
+        ----------
+        path : `str`
+            A relative path to an endpoint.
+        model : ``BaseModel``
```

Comment:
```suggestion
        model : `pydantic.BaseModel`
```
?

Reply:
I thought there was some problem with the Sphinx and the pydantic documentation

Reply:
No, the problem is when you have a model using numpydoc that is inheriting from BaseModel. Sphinx can't render the methods inherited from the base class not using numpydoc. There is no problem with linking in a docstring to a pydantic base model.